### PR TITLE
Refactored HomePage route handler to fetch initClassInfo and getFacil…

### PR DESCRIPTION
**Pull request by shubh1007**
### Summary

This is the continuation of the refactoring process for the coach page, transitioning from a global route handler to local route handlers. #11219

#### List of all planRoutes:

| Route name                  | Handler/component to refactor | Refactor done? | How to test the page                                       |
|-----------------------------|-------------------------------|----------------|-----------------------------------------------------------|
| PageNames.HOME_PAGE         | showHomePage      | yes              | Coach -> Class Home                                              |
| HomeActivityPage.name             | showHomePage                | yes            | Coach -> Class Home -> activity                                    |

### Reviewer guidance

Please follow the table for reviewing the refactored routes.

### Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

### Reviewer checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR is fully functional
- [ ] PR has been tested for accessibility regressions
- [ ] Documentation is updated
- [ ] Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/app.js
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/app.js]: Since the related route uses `PageNames.HOME_PAGE` name, I think it'd be safer to use it here as well (it's possible they're same but as far as I can say that'd be rather a coincidence.

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/routes/index.js:71]: I think that here `async` could be removed because in this function we're not `await`ing, or was there some trouble without it? Same on line 87.

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/routes/index.js]: `notLoading` is now called twice for each of the routes being refactored - I'd suggest to remove it from here, or from lines 78 and 89.

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (MisRob) [inline kolibri/plugins/coach/assets/src/routes/index.js]: Please remove `console.log`. I am not sure why lint check didn't detect it.

**Comment by MisRob** [comment]
Hi @shubh1007, this is great! I can see you grasped the problem and the strategy for solving it, thanks a lot. This will really help us to unblock some important work. Leaving just a few details, but overall looks really good.

Do you wish to continue with some other routes from this file? If so, would you push them to this PR as well or would you rather work in a new one?

Note that I'm mostly offline until July 8 - just chiming in on this issue to confirm that you can continue if you'd like, and on some days I will be available to answer questions.

**Comment by MisRob** [comment]
@shubh1007 I don't think you're on Slack yet, but recently I approached the community of developers with this task and since it's pretty abstract, I also wrote something about what it will have impact on in the near future, if that'd be of your interest:

Screenshot from 2024-06-25 20-49-22

And also it showed up that the upcoming 0.18 Kolibri version will benefit from these updates to unblock some major improvements to the navigation UX. So thanks a lot, despite this is not so visible at this point, coaches using Kolibri will notice it soon :) !

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (shubh1007) [reply kolibri/plugins/coach/assets/src/routes/index.js]: I have removed it.

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (shubh1007) [reply kolibri/plugins/coach/assets/src/routes/index.js]: Resolved.

File: kolibri/plugins/coach/assets/src/routes/index.js
Comment (shubh1007) [reply kolibri/plugins/coach/assets/src/routes/index.js:71]: Yes I tried this approach but it was not able to load the promises properly. So, I added async here.

Comment (MisRob) [reply kolibri/plugins/coach/assets/src/routes/index.js:71]: I am not sure if I can understand what exactly you mean "it was not able to load the promises properly".

If I remove `async` from here, it doesn't throw any errors and works like before in the current code, because we're not calling `await showHomePage(toRoute)`. However, because of that missing `await`, I don't think that promises get resolved before `'notLoading'`.

Overall looking at the way you implemented it, I believe the intention actually was to wait for these two promises to be finished before calling `'notLoading'`, which we indeed need here. To achieve that, we could:

```
function showHomePage(toRoute) {
  const initClassInfoPromise = store.dispatch('initClassInfo', toRoute.params.classId);
  const getFacilitiesPromise =
    store.getters.isSuperuser && store.state.core.facilities.length === 0
      ? store.dispatch('getFacilities').catch(() => {})
      : Promise.resolve();

  return Promise.all([initClassInfoPromise, getFacilitiesPromise]);
}
```

and then

```
{
    name: PageNames.HOME_PAGE,
    path: '/:classId?/home',
    component: HomePage,
    handler: async (toRoute, fromRoute, next) => {
      if (classIdParamRequiredGuard(toRoute, HomePage.name, next)) {
        return;
      }
      await showHomePage(toRoute);
      store.dispatch('notLoading');
    },
```

Is this what you've meant to achieve?

Comment (MisRob) [reply kolibri/plugins/coach/assets/src/routes/index.js:71]: I think this is the last thing to cleanup - seems to be caused by playing around with two possible placements of `store.dispatch('notLoading');` that got mixed up. Since now it's removed from the helper `showHomePage` function, we need to update the structure a bit.

**Comment by MisRob** [comment]
Thanks for following up @shubh1007. I will finish review when I'm back online next week. Meanwhile, I feel confident that we're on the same page about this task so feel free to continue working on other routes if you'd like (or please message us if you don't want to work on more so we know we should re-assign)

**Review by MisRob — APPROVED** [review]
@shubh1007 Thank you. I pushed two last tweaks. See commit messages to understand. As soon as checks pass, we will merge.

**Pull request merged**
